ToDoList/app/src/main/java/com/example/jong/todolist/MainActivity.java

```diff
@@ -0,0 +1,134 @@
+                    public void onClick(DialogInterface dialog, int which) {
+                        String entry1 = textinput.getText().toString();
+                        Log.d("MainActivity", entry1);
+                        arraylist.set(pos, entry1);
+                        ArrayAdapter<String> todolist_db = (ArrayAdapter<String>) listView.getAdapter();
+                        todolist_db.notifyDataSetChanged();
+                    }
+                });
+                edittaskalert.setNegativeButton("Cancel",null);
+                edittaskalert.create().show();
+            }
+        });
+
+            /* Long Click to delete*/
+            listView.setOnItemLongClickListener(new AdapterView.OnItemLongClickListener() {
+
+            @Override
+                public boolean onItemLongClick(AdapterView<?> parent, View view, int position, long id) {
+                final int pos=position;
+                String entry = arraylist.get(position);
+                AlertDialog.Builder deltaskalert = new AlertDialog.Builder(MainActivity.this);
+                deltaskalert.setTitle("Delete the task");
+                deltaskalert.setMessage("Do you really want to delete this task?");
+                deltaskalert.setPositiveButton("Yes", new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which){
+                        Log.d("MainActivity", "Deleting task");
+                        arraylist.remove(pos);
+                        ArrayAdapter<String> todolist_db = (ArrayAdapter<String>) listView.getAdapter();
+                        todolist_db.notifyDataSetChanged();
+                    }
+
+                });
+                deltaskalert.setNegativeButton("Cancel",null);
+                deltaskalert.create().show();
+                return true;
+            }
+        });
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        // Inflate the menu; this adds items to the action bar if it is present.
+        getMenuInflater().inflate(R.menu.menu, menu);
+        return true;
+    }
+
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        // Handle action bar item clicks here. The action bar will
+        // automatically handle clicks on the Home/Up button, so long
+        // as you specify a parent activity in AndroidManifest.xml.
+        int id = item.getItemId();
+
+        //noinspection SimplifiableIfStatement
+         if (id == R.id.action_add_task) {
+
+             AlertDialog.Builder addtaskalert = new AlertDialog.Builder(this);
+             addtaskalert.setTitle("Add a Task");
+             addtaskalert.setMessage("Need to do:");
+             final EditText textinput = new EditText(this);
+             addtaskalert.setView(textinput);
+             DialogInterface.OnClickListener Addbuttonlistener = new DialogInterface.OnClickListener() {
```

Comment:
You could also define this in a separate class like I mentioned in earlier comments